File: mobile/src/main/kotlin/de/scout/fireplace/home/EventMatcher.java
package de.scout.fireplace.home;

import de.scout.fireplace.models.Expose;
import de.scout.fireplace.preference.PreferenceRepository;
import de.scout.fireplace.ui.FloatingCardStackEvent;
import javax.inject.Inject;

class EventMatcher {

  private static final String SPLITTER = "\\|";

  private final PreferenceRepository repository;

  @Inject
  EventMatcher(PreferenceRepository repository) {
    this.repository = repository;
  }

  boolean match(FloatingCardStackEvent event) {
    return event.getType() == FloatingCardStackEvent.Type.APPROVED && matches(event.getSummary());
  }

  private boolean matches(Expose.Summary summary) {
    String[] attributes = summary.getDescription().split(SPLITTER);
    if (attributes.length != 3) {
      return false;
    }

    float price = parseFloat(attributes[0]);
    if (price < repository.getMinPrice() || price > repository.getMaxPrice()) {
      return false;
    }

    int rooms = parseInt(attributes[2]);
    if (rooms < repository.getMinRooms()) {
      return false;
    }

    return true;
  }

  private float parseFloat(String string) {
    return Float.parseFloat(string.replaceAll("[^\\d]", ""));
  }

  private int parseInt(String string) {
    return Integer.parseInt(string.replaceAll("[^\\d]", ""));
  }
}

File: mobile/src/main/kotlin/de/scout/fireplace/home/EventMatcher.kt
package de.scout.fireplace.home

import de.scout.fireplace.models.Expose
import de.scout.fireplace.settings.SettingsRepository
import de.scout.fireplace.ui.FloatingCardStackEvent
import javax.inject.Inject

internal class EventMatcher @Inject constructor(private val repository: SettingsRepository) {

  fun match(event: FloatingCardStackEvent): Boolean {
    return event.type === FloatingCardStackEvent.Type.LIKE && matches(event.expose)
  }

  private fun matches(expose: Expose): Boolean {
    val attributes = expose.attributes
    if (attributes.size != 3) {
      return false
    }

    val price = attributes[0].value.toNumeric().toFloat()
    if (price < repository.minPrice || price > repository.maxPrice) {
      return false
    }

    val rooms = attributes[2].value.toNumeric().toInt()
    if (rooms < repository.minRooms) {
      return false
    }

    return true
  }

  private fun String.toNumeric(): String {
    return replace("[^\\d]".toRegex(), "")
  }
}
